### PR TITLE
Selective timelines

### PR DESCRIPTION
**Pull request by bart-de-jonge**
Added selective timelines

### CODE REVIEW

File: objects/XMLHelper.js [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+                        result.scriptingProject["camera-centerarea"][0].cameraTimeline;
+
+                    const cameraTimelines = [];
+                    const flattenedTimelines = [];
+
+                    // Insert shots in timeline which is pushed to timelinesarray
+                    // and push to flattenedArray
+                    cameraTimelinesXML.forEach((timeline) => {
+                        // Get camera
+                        const camera = this.getCamera(timeline.camera[0]);
```
Comment (geenen124) [inline objects/XMLHelper.js]: Please replace this with a call to the `fromXML(XMLObject)` static method

File: objects/XMLHelper.js [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+import Camera from "./Camera";
+
+const parser = new xml2js.Parser();
+
+/*
+ * Class for storing a CameraTimeline
+ */
+class XMLHelper {
+
+    getCameraType(XMLObject) {
```
Comment (geenen124) [inline objects/XMLHelper.js]: This is a duplicate of the `fromXML(XMLObject)` method in CameraType

File: objects/XMLHelper.js [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+        return new CameraType(
+            XMLObject.name[0],
+            XMLObject.description[0],
+            XMLObject.movementMargin[0]
+        );
+    }
+
+    getCamera(XMLObject) {
```
Comment (geenen124) [inline objects/XMLHelper.js]: This is a duplicate of the `fromXML(XMLObject)` method in Camera

File: routes/index.js [hunk 4]
```diff
@@ -6,4 +6,8 @@ router.get("/", (req, res) => {
     res.render("index");
 });
 
+// router.get("/user-data", (req, res) => {
```
Comment (geenen124) [inline routes/index.js]: Please remove commented code or provide a reason as to why this is still in the repo (e.g. by adding a TODO)

File: views/index.ejs [hunk 5]
```diff
@@ -1,13 +1,24 @@
+
 <!DOCTYPE html>
 <html>
+<head>
+    <% include ./partials/head %>
+    <script src="http://code.jquery.com/jquery-latest.min.js" type="text/javascript"></script>
```
Comment (geenen124) [inline views/index.ejs]: This was removed in a previous commit? Doesn't seem to be used anywhere.

File: views/index.ejs [hunk 6]
```diff
@@ -1,13 +1,24 @@
...
-
+  <body class="layout vertical">
 
     <h1>Welcome to the webserver of goto fail;</h1>
     <p>This is a temporary page and will be replaced with a page that has actual functionality when we have developed these functionalities.</p>
-    <img src="/images/gotofail.png"/>
+
+    <timeline-picker class="flex"></timeline-picker>
+
+    <!--<img src="/images/gotofail.png"/>-->
```
Comment (geenen124) [inline views/index.ejs]: Please remove commented code.

File: elements/timeline-picker.html [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+
+
+<dom-module id="timeline-picker">
+    // Template
+    <template>
+        <style include="iron-flex"></style>
+        <style>
+            paper-card {
+                background-color: #00B7FF;
+                /*width: 550px;*/
```
Comment (geenen124) [inline elements/timeline-picker.html]: Please remove commented code.

File: elements/timeline-picker.js [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+        const checkedToggles = [];
+
+        for (let i = 0; i < toggles.length; i++) {
+            if (toggles[i].checked) {
+                checkedToggles.push(toggles[i].value);
+            }
+        }
+
+        // eslint-disable-next-line
+        $.post("/timeline/picked-timelines", {"pickedTimelines": checkedToggles}, (data) => {
```
Comment (geenen124) [inline elements/timeline-picker.js]: This can be replaced with a Polymer iron-ajax post request

**Comment by geenen124** [comment]
Looks good!

**Comment by Markvanderuit** [comment]
Looks good, merging.

**Pull request merged**